js/ui/marker.js

```diff
@@ -86,36 +88,37 @@ Marker.prototype = {
     },
 
     getElement: function () {
-        return this._el;
+        return this._element;
     },
 
     /**
      * Binds a Popup to the Marker
-     * @param {Popup=} popup an instance of the `Popup` class. If undefined or null, any popup 
+     * @param {Popup=} popup an instance of the `Popup` class. If undefined or null, any popup
      * set on this `Marker` instance is unset
      * @returns {Marker} `this`
      */
 
     setPopup: function (popup) {
-        if (popup == null) {
-            this._closePopup();
-            delete this._popupHandlersAdded;
```

Comment:
> Instead of tracking state in `_popupHandlersAdded`, call`removeEventListener` when the popup is removed. This will prevent memory leaks and unexpected behavior if the `Marker` continues to be used without a popup. You'll need to save the `this`-bound version of `_openPopup`, say by using `util.bindAll(['_openPopup'], this)`, so that it can be passed to `removeEventListener`. -- https://github.com/mapbox/mapbox-gl-js/pull/3056#discussion_r76332110